connectors/kafka/src/main/java/forklift/controller/KafkaController.java

```diff
@@ -145,32 +114,20 @@ public void start() {
      * @throws InterruptedException if the thread is interrupted
      */
     public void stop(long timeout, TimeUnit timeUnit) throws InterruptedException {
-        kafkaConsumer.wakeup();
-        executor.shutdownNow();
-
         running = false;
+        kafkaConsumer.wakeup();
+       // executor.shutdownNow();
+        executor.shutdown();
```

Comment:
Do we even need `shutdown` here? The `kafkaConsumer.wakeup()` call should be sufficient to make the message loop thread terminate.

The difference between the two shutdown methods is that `shutdown` prevents incoming tasks from being queued on the executor (which shouldn't matter here since you only call `start` once on a `KafkaController`), while `shutdown` sends a signal `Thread.interrupt` to it's threads to let them know that they _should_ quit soon.